wpilibNewCommands/src/test/native/cpp/frc2/command/CommandScheduleTest.cpp

```diff
@@ -102,6 +106,43 @@ TEST_F(CommandScheduleTest, NotScheduledCancel) {
   EXPECT_NO_FATAL_FAILURE(scheduler.Cancel(&command));
 }
 
+TEST_F(CommandScheduleTest, CancelNextCommand) {
+  CommandScheduler scheduler = GetScheduler();
+
+  frc2::RunCommand** command1PtrPtr = new frc2::RunCommand*;
+  frc2::RunCommand** command2PtrPtr = new frc2::RunCommand*;
+  auto counterPtr = std::make_shared<int>(0);
+
+  auto command1 = frc2::RunCommand([counterPtr, &command2PtrPtr, &scheduler] {
+    scheduler.Cancel(*command2PtrPtr);
+    (*counterPtr)++;
+  });
+  auto command2 = frc2::RunCommand([counterPtr, &command1PtrPtr, &scheduler] {
+    scheduler.Cancel(*command1PtrPtr);
+    (*counterPtr)++;
+  });
+
+  *command1PtrPtr = &command1;
+  *command2PtrPtr = &command2;
```

Comment:
This can be made clearer by using capture-by-reference in the lambdas. (This also avoids a memory leak from never freeing the pointers allocated by `new`)
```suggestion
  frc2::RunCommand* command1Ptr = nullptr;
  frc2::RunCommand* command2Ptr = nullptr;
  int counter = 0;

  auto command1 = frc2::RunCommand([&counter, &command2Ptr, &scheduler] {
    scheduler.Cancel(command2Ptr);
    counter++;
  });
  auto command2 = frc2::RunCommand([&counter, &command1Ptr, &scheduler] {
    scheduler.Cancel(command1Ptr);
    counter++;
  });

  command1Ptr = &command1;
  command2Ptr = &command2;
```
You'll also need to update the other usages of `counterPtr`.
I think you might find it helpful to think of references as being const pointers (NOT pointers to const) that are automatically dereferenced. Hopefully that explains why you can have the variables store what data it needs (which may be reassigned later) and have the lambdas capture by reference so that when the variable gets a different value, the captured value is also updated.

Reply:
Thx, finished doing the test at 5 AM where I just wanted to see it work, I will update it to this version with a rebase asap.
For some reason, doing a pass-by-reference didn't work last time, so I did it this way, but let me check again. I may add I'm not used to C++ lambdas so that's why I did that kind of double pointer way. 

Also, IMO the shared pointer makes it easier to debug so I will keep it that way.

Reply:
added into it with rebase see: 7149e43